File: src/main/java/org/intellij/sdk/language/jass/JassColorSettingsPage.java
package org.intellij.sdk.language.jass;

import com.intellij.openapi.editor.colors.TextAttributesKey;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.options.colors.AttributesDescriptor;
import com.intellij.openapi.options.colors.ColorDescriptor;
import com.intellij.openapi.options.colors.ColorSettingsPage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.util.Map;

final class JassColorSettingsPage implements ColorSettingsPage {

    private static final AttributesDescriptor[] DESCRIPTORS = new AttributesDescriptor[]{
            new AttributesDescriptor("Key", JassSyntaxHighlighter.KEY),
            new AttributesDescriptor("Separator", JassSyntaxHighlighter.SEPARATOR),
            new AttributesDescriptor("Value", JassSyntaxHighlighter.VALUE),
            new AttributesDescriptor("Bad value", JassSyntaxHighlighter.BAD_CHARACTER)
    };

    @Override
    public @NotNull Icon getIcon() {
        return JassIcons.FILE;
    }

    @NotNull
    @Override
    public SyntaxHighlighter getHighlighter() {
        return new JassSyntaxHighlighter();
    }

    @NotNull
    @Override
    public String getDemoText() {
        return """
                # You are reading the ".properties" entry.
                ! The exclamation mark can also mark text as comments.
                website = https://en.wikipedia.org/
                language = English
                # The backslash below tells the application to continue reading
                # the value onto the next line.
                message = Welcome to \\
                          Wikipedia!
                # Add spaces to the key
                key\\ with\\ spaces = This is the value that could be looked up with the key "key with spaces".
                # Unicode
                tab : \\u0009""";
    }

    @Nullable
    @Override
    public Map<String, TextAttributesKey> getAdditionalHighlightingTagToDescriptorMap() {
        return null;
    }

    @Override
    public AttributesDescriptor @NotNull [] getAttributeDescriptors() {
        return DESCRIPTORS;
    }

    @Override
    public ColorDescriptor @NotNull [] getColorDescriptors() {
        return ColorDescriptor.EMPTY_ARRAY;
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "Jass";
    }

}

File: src/main/kotlin/raft/war/language/jass/JassColorSettingsPage.kt
package raft.war.language.jass

import com.intellij.openapi.editor.colors.TextAttributesKey
import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.options.colors.AttributesDescriptor
import com.intellij.openapi.options.colors.ColorDescriptor
import com.intellij.openapi.options.colors.ColorSettingsPage
import raft.war.language.jass.JassIcons.FILE
import raft.war.language.jass.highlighter.JassSyntaxHighlighterBase
import javax.swing.Icon

/*
Юзер функции - жёлтые.
Нативки - розовые/фиолетовые.
БЖ - красные.
 */

internal class JassColorSettingsPage : ColorSettingsPage {
    override fun getIcon(): Icon = FILE

    override fun getHighlighter(): SyntaxHighlighter = JassSyntaxHighlighterBase()

    override fun getDemoText(): String = """
            ${'$'} // bad character
            type <TYPE_NAME>agent</TYPE_NAME> extends <TYPE_NAME>handle</TYPE_NAME>
            
            """.trimIndent()

    override fun getAdditionalHighlightingTagToDescriptorMap(): HashMap<String, TextAttributesKey?> =
        object : HashMap<String, TextAttributesKey?>() {
            init {
                put("TYPE_NAME", JassSyntaxHighlighterBase.JASS_TYPE_NAME)
            }
        }

    override fun getAttributeDescriptors(): Array<AttributesDescriptor> = arrayOf(
        AttributesDescriptor("Identifier", JassSyntaxHighlighterBase.JASS_ID),
        AttributesDescriptor("Type", JassSyntaxHighlighterBase.JASS_TYPE_NAME),
        AttributesDescriptor("Keyword", JassSyntaxHighlighterBase.JASS_KEYWORD),
        AttributesDescriptor("Comment", JassSyntaxHighlighterBase.JASS_LINE_COMMENT),
        AttributesDescriptor("Bad character", JassSyntaxHighlighterBase.JASS_BAD_CHARACTER),
    )

    override fun getColorDescriptors(): Array<ColorDescriptor> = ColorDescriptor.EMPTY_ARRAY

    override fun getDisplayName(): String = JassLanguage.NAME
}

